crates/codegen/parser/runtime/src/parse_error.rs

```diff
@@ -11,6 +11,21 @@ pub struct ParseError {
     pub(crate) tokens_that_would_have_allowed_more_progress: Vec<TokenKind>,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum WithColor {
```

Comment:
A bit of history on this:

We eventually will only expose typed errors, with a `impl Display` to stringify it.
The `with_color: true` use case is only used in our CLI, and is not very useful for others right now.

In the meantime, instead of exposing another public type that we are going to deprecate, should we just have this as an internal overload for our CLI, and remove this argument from the public API and NAPI (passing `false`)?

Reply:
If we aim to keep it as a public set of modular compiler APIs, I wouldn't think it's uncommon to be able to enable/disable coloring for the error reports even for our public consumers. If we consider some parts of our API unstable, maybe it'd be good to add some docs saying that we discourage using something for whatever reason?

I'll revert this change so that it's only about fixing the lints, then.